.github/workflows/component-shared-workflow.yml

```
@@ -0,0 +1,103 @@
+# .github/workflows/component-shared-workflow.yml
+
+name: Component Build
+
+on:
+  workflow_call:
+    inputs:
+      component:
+        required: true
+        type: string
+      OS:
+        required: true
+        type: string
+      skipFilter:
+        required: false
+        type: string
+      runConfigServer:
+        required: false
+        type: boolean
+        default: false
+jobs:
+  build:
+    name: Build ${{ inputs.component }} on ${{ inputs.OS }}
+    runs-on: ${{ inputs.OS }}-latest
+
+    services:
+      eurekaServer:
+        image: ${{ inputs.runConfigServer && 'steeltoe.azurecr.io/eureka-server' || null }}
+        ports:
+          - 8761:8761
+      configServer:
+        image: ${{ inputs.runConfigServer && 'steeltoe.azurecr.io/config-server' || null }}
+        env:
+          eureka.client.enabled: true
+          eureka.client.serviceUrl.defaultZone: http://eurekaServer:8761/eureka
+          eureka.instance.hostname: localhost
+          eureka.instance.instanceId: localhost:configServer:8888
+        ports:
+          - 8888:8888
+
+    env:
+      DOTNET_NOLOGO: true
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+      DOTNET_GENERATE_ASPNET_CERTIFICATE: ${{ inputs.OS == 'macOS' && 'false' || '' }}
+      SolutionFile: src/Steeltoe.${{ inputs.component }}.slnf
+      CommonTestArgs: >-
+        --no-build --configuration Release --blame-crash --blame-hang-timeout 3m
+        --collect "XPlat Code Coverage" --settings coverlet.runsettings --logger trx --results-directory ${{ github.workspace }}
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: |
+            8.0.*
+            9.0.*
+
+      - name: Install Nerdbank.GitVersioning (macOS only)
+        if: ${{ inputs.OS == 'macos' }}
+        run: dotnet tool install --global nbgv
+
+      - name: Set package version
+        run: nbgv cloud
+
+      - name: Restore dependencies
+        run: dotnet restore ${{ env.SolutionFile }} --verbosity minimal --configfile nuget.config
```

```suggestion
        run: dotnet restore ${{ env.SolutionFile }} --verbosity minimal
```

Should automatically be picked up.